DOC: Corrects 'reindex_axis' docstring

eahogue opened the pull request:
The docstring included a 'tolerance' parameter which is missing from the source. This has been removed.

The docstring did not describe the 'fill_value' parameter. This has been added, mostly reusing the wording in the summary part of the existing docstring.

- [x] closes #23960
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

eahogue committed:
Corrects 'reindex_axis' docstring.

The docstring included a 'tolerance' parameter which is missing from the source. This has been removed.

The docstring did not describe the 'fill_value' parameter. This has been added, mostly reusing the wording in the summary part of the existing docstring.

eahogue committed:
Merge pull request #1 from eahogue/eahogue-patch-1

Corrects 'reindex_axis' docstring.

pep8speaks commented:
Hello @eahogue! Thanks for submitting the PR.

 - In the file [`pandas/core/generic.py`](https://github.com/eahogue/pandas/blob/1208e50ca0ad336fc742e78531e1c6e79d121fab/pandas/core/generic.py), following                are the PEP8 issues :
> [Line 4342:80](https://github.com/eahogue/pandas/blob/1208e50ca0ad336fc742e78531e1c6e79d121fab/pandas/core/generic.py#L4342): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)

eahogue committed:
Minor PEP-8 change to previous commit

Shortened description line of 'fill_value' in 'reindex_axis'.

datapythonista commented:
Thanks for the fix @eahogue 

Do you mind running `./scripts/validate_docstrings.py pandas.DataFrame.reindex_axis` to see if there is any other error.

I think a short summary is missing (a one liner summary), the `Returns` doesn't have a description, and the example is missing the output.

If you don't mind fixing those, and anything else reported by the script, or that you see, that would be great. So we leave this docstring perfect.

eahogue committed:
Fixed non-functioning example.
Added periods, reformatted summary, and other changes required by
validate_docstrings.py.

eahogue commented:
Thanks, @datapythonista. I have made additional changes and hope I have committed etc correctly. The validation script no longer returns errors. PTAL.

datapythonista reviewed (commented):
Thanks @eahogue 

Looking much better, I added some final comments. If you don't mind addressing those too, and I think we're ready to go.

datapythonista committed:
Update pandas/core/generic.py

Co-Authored-By: eahogue <eahogue@gmail.com>

datapythonista committed:
Update pandas/core/generic.py

Co-Authored-By: eahogue <eahogue@gmail.com>

datapythonista committed:
Update pandas/core/generic.py

Co-Authored-By: eahogue <eahogue@gmail.com>

eahogue commented on pandas/core/generic.py:
```diff
@@ -4311,60 +4311,63 @@ def _needs_reindex_multi(self, axes, method, level):
 
+        Returns
+        -------
+        %(klass)s
```
Done.

eahogue commented on pandas/core/generic.py:
```diff
@@ -4311,60 +4311,63 @@ def _needs_reindex_multi(self, axes, method, level):
         Examples
         --------
-        >>> df.reindex_axis(['A', 'B', 'C'], axis=1)
+        >>> df = pd.DataFrame(np.array(([1,2,3], [4,5,6], [7,8,9])),
```
Done. (The script did not report it, though, fyi.)

eahogue committed:
Mostly stylistic updates to reindex_axis docstring.

eahogue commented on pandas/core/generic.py:
```diff
@@ -4311,60 +4311,63 @@ def _needs_reindex_multi(self, axes, method, level):
-        >>> df.reindex_axis(['A', 'B', 'C'], axis=1)
+        >>> df = pd.DataFrame(np.array(([1,2,3], [4,5,6], [7,8,9])),
+        ...                             index=['One', 'Two', 'Three'],
+        ...                             columns=['A', 'B', 'C'])
```
Done.

eahogue commented on pandas/core/generic.py:
```diff
@@ -4311,60 +4311,63 @@ def _needs_reindex_multi(self, axes, method, level):
+        >>> df = pd.DataFrame(np.array(([1,2,3], [4,5,6], [7,8,9])),
+        ...                             index=['One', 'Two', 'Three'],
+        ...                             columns=['A', 'B', 'C'])
+        >>> df.reindex_axis(['B', 'C', 'D'], axis=1)
```
Done.

eahogue commented on pandas/core/generic.py:
```diff
@@ -4311,60 +4311,63 @@ def _needs_reindex_multi(self, axes, method, level):
+               B  C   D
+        One    2  3 NaN
+        Two    5  6 NaN
+        Three  8  9 NaN
```
Done.

eahogue reviewed (commented):
@datapythonista ptal

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -4311,60 +4311,67 @@ def _needs_reindex_multi(self, axes, method, level):
+        By default, places NaN in locations having no value in the
+        previous index. A new object is produced unless the new index
+        is equivalent to the current one and copy=False.
 
```
`reindex_axis` is a deprecated method, so it would be good to add a 

```
.. deprecated:: 0.21.0
    Use `reindex` instead.

eahogue replied:
Thanks, I have added this to the end of the summary. Based on other docstrings it looks like that's the place to put it.

jorisvandenbossche replied:
Yes, that looks good!

eahogue committed:
Adding deprecation note to reindex_axis docstring.

jreback commented:
can you merge master

datapythonista commented:
@eahogue can you fix the conflicts

Also, the deprecation directive should be places between the short and the extended summary, seems like we're doing it wrong in most places: https://numpydoc.readthedocs.io/en/latest/format.html#sections

I create #24143to fix the remaining errors.

eahogue committed:
Merge branch 'master' into master

eahogue committed:
Minor lint fixes for line length and trailing spaces.

eahogue committed:
Merge branch 'master' of https://github.com/eahogue/pandas

eahogue commented:
Sorry for the basic question, but the tests have passed but I am not sure what else needs to be done. Have the changes gone through yet?

jorisvandenbossche committed:
move deprecated

jorisvandenbossche commented:
I think all comments were addressed, so nothing needed to be done any more (except someone merging it).

@eahogue Thanks a lot!
